duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -354,12 +354,24 @@ private String canonQueryString()
           .replace("+", "%20")
           .replace("*", "%2A")
           .replace("%7E", "~");
-      String value = URLEncoder
-          .encode(params.get(key).toString(), "UTF-8")
-          .replace("+", "%20")
-          .replace("*", "%2A")
-          .replace("%7E", "~");
-      args.add(name + "=" + value);
+      if (params.get(key) instanceof List) {
+        List<String> value_list = (List<String>) params.get(key);
```

Comment:
I am not sure if this is the javaic way (or even if that's the right word) to do this.